src/composables/useGetParticipants.js

```diff
@@ -0,0 +1,152 @@
+
+	/**
+	 * Initialise the get participants functionality
+	 *
+	 */
+	function initialiseGetParticipants() {
+		EventBus.$on('route-change', onRouteChange)
+		EventBus.$on('joined-conversation', onJoinedConversation)
+
+		// FIXME this works only temporary until signaling is fixed to be only on the calls
+		// Then we have to search for another solution. Maybe the room list which we update
+		// periodically gets a hash of all online sessions?
+		EventBus.$on('signaling-participant-list-changed', debounceUpdateParticipants)
+		subscribe('guest-promoted', onJoinedConversation)
+	}
+
+	/**
+	 * Stop the get participants functionality
+	 *
+	 */
+	function stopGetParticipants() {
+		EventBus.$off('route-change', onRouteChange)
+		EventBus.$off('joined-conversation', onJoinedConversation)
+		EventBus.$off('signaling-participant-list-changed', debounceUpdateParticipants)
+		unsubscribe('guest-promoted', onJoinedConversation)
+	}
+
+	const onRouteChange = () => {
+		participantsInitialised.value = store.getters.participantsList(token.value).length > 1
+	}
+
+	const onJoinedConversation = () => {
+		if (isOneToOneConversation.value) {
+			cancelableGetParticipants()
+		} else {
+			nextTick(() => debounceUpdateParticipants())
+		}
+	}
+
+	const debounceUpdateParticipants = () => {
+		if (!isActive.value && !isInCall.value) {
+			// Update is ignored but there is a flag to force the participants update
+			pendingChanges = true
+			return
+		}
+
+		if (store.getters.windowIsVisible() && (isInCall.value || !conversation.value?.hasCall)) {
+			debounceFastUpdateParticipants()
+		} else {
+			debounceSlowUpdateParticipants()
+		}
+	    pendingChanges = false
+	}
+
+	const cancelableGetParticipants = async () => {
+		const isInLobby = store.getters.isInLobby
+		const isModeratorOrUser = store.getters.isModeratorOrUser
+		if (fetchingParticipants || token.value === '' || isInLobby || !isModeratorOrUser) {
+			return
+		}
+
+		fetchingParticipants = true
+		debounceFastUpdateParticipants.clear()
+		debounceSlowUpdateParticipants.clear()
+
+		const response = await store.dispatch('fetchParticipants', { token: token.value })
+		if (response) {
+			participantsInitialised.value = true
+		}
+		fetchingParticipants = false
+	}
+
+	const debounceFastUpdateParticipants = debounce(
+		cancelableGetParticipants, 3000)
+	const debounceSlowUpdateParticipants = debounce(
+		cancelableGetParticipants, 15000)
+
+	// Group conversations have composable in RightSidebar, so should work only for one-to-one
+	watch(isOneToOneConversation, (newValue) => {
+		if (newValue) {
+			initialiseGetParticipants()
+		} else {
+			stopGetParticipants()
+		}
+	}, { immediate: true })
+
+	watch(isActive, (newValue) => {
+		if (newValue && pendingChanges) {
+			debounceUpdateParticipants()
+		}
+	})
+
+	onBeforeUnmount(
+		() => {
+			stopGetParticipants()
+		}
+	)
+
+	return {
+		participantsInitialised,
+		initialiseGetParticipants,
+		cancelableGetParticipants,
+	}
```

Comment:
`initialiseGetParticipants` is not needed exposed as well as executing it in `ParticipantsTab`. It is executed on created anyway by the immediate watch above.

Running it manually in the component runs it a second time on the instance.

Reply:
The immediate watch is for one to one conversations only. Group conversations have the composable initialized in `ParticipantsTab` and gets stopped when it is unmounted, one to one conversations have the composable initialized in `TopBar` and gets stopped as soon as you change the conversation to other then one to one. 

It can't be called twice because participants tab is not mounted when isOneToOneConversation is truthy so there is no overlap. 
The only thing that will be called twice is `stopGetParticipants `when going from one to one conv to group conv because `TopBar `and `ParticipantsTab` are both mounted and they execute the immediate watch in case of `!isOneToOneConversation`.

I think the only way to avoid the duplicated call is to move `stopGetParticipants` back to `TopBar` 🤔 .

Reply:
Or we consider keeping a composable in TopBar only, as it's always mounted for all types of conversations. Then we won't need to invent and care about some conditions to run it in one place and stop in another

Reply:
GetParticipants needs the composable ( `participantsInitialised` and `cancelableGetParticipants`)

Reply:
Right, but that's only readonly ref and fetch request. So we could keep composable in both, but:
- enable/disable listeners only in TopBar
- migrate and modify sidebarStore, so it contains value of activeTab (then we could use it for fast/slow update)
- detach initialization logic from sidebar, keep only `participantsInitialised` and `cancelableGetParticipants`

Reply:
> migrate and modify sidebarStore, so it contains value of activeTab (then we could use it for fast/slow update)

I don't want to make the pace of update dependent on sidebar because this implies that fast update is limited to only when participants tab is open. One-to-one conversation avatar update will be easily noticeable that it is slow. 

For the migration, sure but follow-up :).